src/shared/geom.h

```diff
@@ -77,7 +77,7 @@ struct vec
         float v[3];
     };
 
-    vec() {}
+    vec() : x(0), y(0), z(0) {}
```

Comment:
You should rather initialize all three directly next to their declaration instead of using the constructor. This way, if there ever is the need for additional constructors, these values will have a sane default.

It is usually a good idea to do this for all integral types (e.g., numeric types, pointers (`= nullptr`), etc.

Reply:
Sure